### ISSUE TITLE
Different versions not saved in google storage

### ISSUE THREAD

**Issue by LeKristapino** [metaware/carrierwave-google-storage#1]
Hi, thanks for the great gem!

I am having an issue, though, with saving thumbnails of my files in storage. It all works well in development, where I use file system but when I try to save page thumbnails through gcloud, it does not seem to be working.

Here's my uploader code:

```ruby
class PageUploader < CarrierWave::Uploader::Base
  # Include RMagick or MiniMagick support:
  # include CarrierWave::RMagick
  include CarrierWave::MiniMagick

  def store_dir
    if Rails.env.production?
      "uploads/pages/#{model.parent_file_id}"
    elsif Rails.env.demo?
      "demo/uploads/pages/#{model.parent_file_id}"
    elsif Rails.env.test?
      "tmp/uploads/pages/#{model.parent_file_id}"
    else
      "development/uploads/pages/#{model.parent_file_id}"
    end
  end

  def extension_white_list
    ['pdf']
  end

  version :thumb do
    begin
      unless Rails.env.test?
        process convert: 'jpg' 
        process resize_to_fit: [595, 842]
        process :set_content_type
      end
      def full_filename(for_file = model.attachment.file)
        "#{model.id}-thumb.jpg"
      end
    rescue Exception => e
      puts e.inspect
      true
    end
  end

  def set_content_type(*args)
    self.file.instance_variable_set(:@content_type, 'image/jepg')
  end

  def thumb_url
    thumb.try(:authenticated_url) || thumb.url
  end
end
```
Any help would be appreciated :)

**Comment by LeKristapino** [comment]
The problem seemed to be overriden `store` method. It did not correctly use `filename` and `content_type` that was set inside the uploader.

This is a solution that worked for me (overwriting store once again):
```
CarrierWave::Storage::GcloudFile.class_eval do
  def store(new_file)
    bucket.create_file new_file.path, path, content_type: new_file.content_type
    self
  end
```

**Comment by jasdeepsingh** [comment]
Thank you for reporting this, I'll take a look at this today.

**Comment by webattitude** [comment]
any news about this ?

**Comment by krtschmr** [comment]
!!!

still not working and a fix/hack isn't so obvious

**Comment by krtschmr** [comment]
@LeKristapino  thank your for your fix!

**Comment by jasdeepsingh** [comment]
sorry guys for making you wait on this. I've released `0.7.0` with this fix. please update your `Gemfile`
